jsettlers.logic/src/main/java/jsettlers/logic/movable/strategies/specialists/PioneerStrategy.java

```diff
@@ -87,22 +89,22 @@ private void findWorkablePosition() {
 		}
 	}
 
-	private final EDirection getCloseForeignTile() {
-		EDirection bestNeighbourDir = null;
-		double bestNeighbourDistance = Double.MAX_VALUE; // distance from start point
+	private EDirection getCloseForeignTile() {
+		EDirection[] bestNeighbourDir = new EDirection[1];
+		double[] bestNeighbourDistance = new double[] { Double.MAX_VALUE }; // distance from start point
 
 		ShortPoint2D position = movable.getPos();
-		for (ShortPoint2D satelitePos : new HexGridArea(position.x, position.y, 1, 6)) {
-
-			if (super.isValidPosition(satelitePos) && canWorkOnPos(satelitePos)) {
-				double distance = Math.hypot(satelitePos.x - centerPos.x, satelitePos.y - centerPos.y);
-				if (distance < bestNeighbourDistance) {
-					bestNeighbourDistance = distance;
-					bestNeighbourDir = EDirection.getApproxDirection(position, satelitePos);
+		HexGridArea.iterate(position.x, position.y, 1, 6, visitor(((currX, currY, radius) -> {
+			ShortPoint2D currPosition = new ShortPoint2D(currX, currY);
+			if (super.isValidPosition(currPosition) && canWorkOnPos(currPosition)) {
+				double distance = Math.hypot(currX - centerPos.x, currY - centerPos.y);
```

Comment:
(1) This is not the "real" distance.
(2) `Math.hypot(a, b)` is slow, much slower than `Math.sqrt(a*a + b*b)`. Use the squared distance to avoid both of them.